libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -854,6 +854,35 @@ def DetachDisk(self, disk: 'GoogleComputeDisk') -> None:
     response = request.execute()
     self.BlockOperation(response, zone=self.zone)
 
+  def Delete(self, delete_disks: bool = False) -> None:
+    """Delete an Instance.
+
+    Args:
+      delete_disks (bool): whether to also force delete all attached disks
+        (ignores the 'readonly' bit).
+    """
+    disks_to_delete = []
+    if delete_disks:
+      disks_to_delete = [
+          disk['source'].split('/')[-1] for disk in self.GetValue('disks')]
+
+    gce_instance_client = self.GceApi().instances()
+    logger.info(
+        self.FormatLogMessage('Deleted Instance: {0:s}'.format(self.name)))
+    request = gce_instance_client.delete(
+        project=self.project_id, instance=self.name, zone=self.zone)
+    response = request.execute()
```

Comment:
add the delete instance in a try/except block to catch the resource not found error.
```suggestion
    try:
      request = gce_instance_client.delete(
          project=self.project_id, instance=self.name, zone=self.zone)
        response = request.execute()
    except HttpError as error:
      if error.resp.status == 404:
        log.warning(('Can not find resource {0:s}, the it might be already '
            'deleted. API call resulted in the following error: '
            '{1:s}').format(self.name, str(error)))
      else:
        log.error(
            'While deleting resource {0:s} the following error occurred: {1:s}'
            .format(self.name, str(error))) 
```

Reply:
Done.
Did the same with the disk.Delete() method